### PR TITLE
Release 0.52

### PR DESCRIPTION
**Pull request by fpetkovski**
## Description

Merges changes from the 0.52.1 release back to the main branch.

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: pkg/prometheus/promcfg.go [hunk 1]
```diff
@@ -683,7 +683,7 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 		relabelings = append(relabelings, yaml.MapSlice{
 			{Key: "action", Value: "keep"},
 			{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_label_" + sanitizeLabelName(k), "__meta_kubernetes_pod_labelpresent_" + sanitizeLabelName(k)}},
-			{Key: "regex", Value: fmt.Sprintf("%s;true", m.Spec.Selector.MatchLabels[k])},
+			{Key: "regex", Value: fmt.Sprintf("(%s);true", m.Spec.Selector.MatchLabels[k])},
```
Comment (fpetkovski) [inline pkg/prometheus/promcfg.go:686]: I wonder why these changes are still shown in the diff. They should already be in the main branch :thinking:

Comment (fpetkovski) [reply pkg/prometheus/promcfg.go:686]: Ok, looks like they were never merged into main.

Comment (paulfantom) [reply pkg/prometheus/promcfg.go:686]: [They are not](https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg.go#L684-L686). The PR introducing this (https://github.com/prometheus-operator/prometheus-operator/pull/4395) was merged to `release-0.52` and from what I see the idea was that we will bring it into `main` during syncing 0.52.1 into `main` (so this PR).

**Review by paulfantom — APPROVED** [review]
lgtm on green
